python-sdk/src/astro/sql/operators/append.py

```diff
@@ -59,8 +62,43 @@ def execute(self, context: Context) -> BaseTable:  # skipcq: PYL-W0613
             target_table=self.target_table,
             source_to_target_columns_map=self.columns,
         )
+
         return self.target_table
 
+    def get_openlineage_facets(self) -> OpenLineageFacets:
+        input_dataset: list[OpenlineageDataset] = [
+            OpenlineageDataset(
+                namespace=self.source_table.openlineage_dataset_namespace(),
+                name=self.source_table.openlineage_dataset_name(),
+                facets={
+                    "table_name": self.source_table.name,
+                    "row_affected": 0,  # FixMe
+                    "columns": self.columns,
+                    "metadata": self.target_table.metadata,
+                },
+            )
+        ]
+
+        output_dataset: list[OpenlineageDataset] = [
+            OpenlineageDataset(
+                namespace=self.target_table.openlineage_dataset_namespace(),
+                name=self.target_table.openlineage_dataset_name(),
+                facets={
+                    "table_name": self.target_table.name,
+                    "row_affected": 0,  # FixMe
+                    "columns": self.columns,
+                    "metadata": self.target_table.metadata,
+                },
+            )
+        ]
+
+        run_facets: dict[str, BaseFacet] = {}
```

Comment:
Do you think this will always be empty in append operator. Job facet can contain sql as mentioned in https://github.com/OpenLineage/OpenLineage/blob/main/spec/OpenLineage.md#run-facets

Reply:
any thought on how we can access sql stmt it get form in base class https://github.com/astronomer/astro-sdk/blob/main/python-sdk/src/astro/databases/base.py#L594 ?

Reply:
You could add `sql` to a class attribute and use it as part of `self.<attribute-name>`

Reply:
As discussed offline we are using xcom for this